### PR TITLE
fix: ignore submodules when looking for out-of-date data

### LINKED ISSUE #714: renku dataset erroneously reports "Migration is required"
**Describe the bug**
If a renku project has a submodule that contains data, `renku dataset` reports that the project needs to be migrated. This is incorrect.

**To Reproduce**
- Add as a submodule in the `data` folder a renku project that contains data

**Expected behavior**
Renku should not consider submodules when determining if datasets need to be migrated.

**Desktop (please complete the following information):**
- renku version: 0.6.1.dev4

### PR DESCRIPTION
**Pull request by ciyer**
# Description

Fix (#714 ) by ignoring data in submodules that are up-to-date.

Fixes #714.

## Type of change

Please select relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

**Do not create pull request unless you checked all points.**

- [x] I have read and understood the **CONTRIBUTING** document.
- [x] I have performed a self-review of my own code.
- [x] I have signed and sent the [CLA document]
      (https://github.com/SwissDataScienceCenter/documentation/wiki/files/SDSC_Contributor_License_Agreement_v1.0.pdf).
- [ ] I have added tests to cover my changes.
- [x] I have **NOT** removed any existing tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have described *all* changes in the **CHANGELOG.rst**.
- [x] I have run ``./run-tests.sh`` locally.

### CODE REVIEW

**Comment by mohammad-sdsc** [comment]
Hi Sekhar,
Thanks for your PR. If you have this problem with Renku projects that their version is `2` then my datasets PR solves this issue by checking the project version and suppressing this error if project is version '2'. See https://github.com/SwissDataScienceCenter/renku-python/pull/710 for more info.

File: conftest.py [hunk 1]
```diff
@@ -30,6 +30,10 @@
 import responses
 from click.testing import CliRunner
 
+import ptvsd
```
Comment (rokroskar) [inline conftest.py]: I don't think we want this commited?

**Comment by jsam** [comment]
Any status on this or can we close the PR?

**Comment by jsam** [comment]
@ciyer I'm closing this PR due to its quite old age. If you decide you still need this please reopen the PR.

**Pull request closed without merging**